GameEngineTests/StochasticDeckTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿using GameEngine;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GameEngineTests
+{
+    [TestClass]
+    public class StochasticDeckTests : DeckTests<StochasticDeck>
+    {
+        protected override StochasticDeck InitializeDeck(int gameSizeMultiplier, int? numberOfSunCards = null)
+        {
+            return new StochasticDeck(gameSizeMultiplier, numberOfSunCards);
+        }
+
+        [TestMethod]
+        public void ShouldProvideAccurateCardCountAfterCardsAreDrawn()
+        {
+            var deck = new StochasticDeck(6);
+            var expectedCardWeights = InitialCardWeights(6);
+            
+            CollectionAssert.AreEquivalent(expectedCardWeights, deck.CardWeights);
+            
+            var drawnCards = deck.Draw(10).ToList();
+            drawnCards.ForEach(card => --expectedCardWeights[card]);
+
+            CollectionAssert.AreEquivalent(expectedCardWeights, deck.CardWeights);
+        }
+
+        [TestMethod]
+        public void ShouldNotJustDrawFromTheTopOfTheDeck()
+        {
+            var deck = new StochasticDeck(1000);
+            var top100Cards = deck.Cards.GetRange(0, 100);
+
+            var actualDraws = deck.Draw(100);
+
+            // It's possible this fails, but it's very, very unlikely.
+            CollectionAssert.AreNotEqual(top100Cards, actualDraws);
```

Comment:
💡 I'm not suggesting you do this, but there are tests for randomness: https://en.wikipedia.org/wiki/Randomness_tests
I think commenting that the vanishingly likely thing could happen is sufficient.

Reply:
Interesting.
Because I'm curious, I decided to figure out how likely this test is to hit a collision.
With a game multiplier of 1000, and the default sun card density of 28%, there are 8333 total cards in the deck. The uneven distribution messes with the numbers a little bit, so this is rough, but since order maters (`AreNotEqual`, not `AreNotEquivalent`) there are roughly 7^100 ways to draw 100 cards. There's only about 1 chance in 7 duotrigintillion that this test gives a false positive. I'm pretty ok with this.